### PR TITLE
feat: Add `optimum-haystack`

### PR DESCRIPTION
**Pull request by shadeMe**

### CODE REVIEW

File: integrations/optimum.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+indexing_pipeline.add_component("embedder", OptimumDocumentEmbedder(
+    model="intfloat/e5-base-v2",
+    normalize_embeddings=True,
+    pooling_mode=OptimumEmbedderPooling.MEAN,
+))
+indexing_pipeline.add_component("writer", DocumentWriter(document_store=document_store))
+indexing_pipeline.connect("embedder", "writer")
+
+indexing_pipeline.run({"embedder": {"documents": documents}})
+```
```
Comment (TuanaCelik) [inline integrations/optimum.md:78]: License section missing 🙏

**Comment by TuanaCelik** [comment]
Can we add the same hugging face logo we already have to this one too?

**Comment by shadeMe** [comment]
> Can we add the same hugging face logo we already have to this one too?

I went for the existing logo as Optimum doesn't appear to have one of its own.

File: integrations/optimum.md (hunk 1 above)
Comment (shadeMe) [reply integrations/optimum.md:79]: Added.

**Review by TuanaCelik — APPROVED** [review]

**Pull request merged**